constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -58,49 +56,61 @@ pub enum Error {
 /// an unknown variable gets known and provides functions to solve
 /// (some kinds of) equations.
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct QuadraticSymbolicExpression<T: FieldElement, V> {
+pub struct QuadraticSymbolicExpressionImpl<T, V> {
     /// Quadratic terms of the form `a * X * Y`, where `a` is a (symbolically)
     /// known value and `X` and `Y` are quadratic symbolic expressions that
     /// have at least one unknown.
     quadratic: Vec<(Self, Self)>,
     /// Linear terms of the form `a * X`, where `a` is a (symbolically) known
     /// value and `X` is an unknown variable.
-    linear: BTreeMap<V, SymbolicExpression<T, V>>,
+    linear: BTreeMap<V, T>,
     /// Constant term, a (symbolically) known value.
-    constant: SymbolicExpression<T, V>,
+    constant: T,
 }
 
-impl<T: FieldElement, V> From<SymbolicExpression<T, V>> for QuadraticSymbolicExpression<T, V> {
-    fn from(k: SymbolicExpression<T, V>) -> Self {
+// TODO: This type is equivalent to a pre-refactoring version of `QuadraticSymbolicExpressionImpl`.
+// It should be removed in a follow-up PR.
+pub type QuadraticSymbolicExpression<T, V> =
+    QuadraticSymbolicExpressionImpl<SymbolicExpression<T, V>, V>;
+
+// Note that we can't also implement `From<T>` for `QuadraticSymbolicExpressionImpl<T, V>`,
```

Comment:
I know this is probably more convenient for the code we are writing at the moment, but in principle, there should rather be a `From<RuntimeConstant>` than a `From<RuntimeConstant::FieldType>`